src/Hidden/HiddenJs.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+/**
+ * Responsively hides children by omission.
+ */
+import React, { Element } from 'react';
+import { keys as breakpoints } from '../styles/breakpoints';
+import withWidth, { isWidthDown, isWidthUp } from '../utils/withWidth';
+import type { Props } from './Hidden';
+import { defaultProps } from './Hidden';
+
+type JsProps = Props & {
+  /**
+   * @ignore
+   * width prop provided by withWidth decorator
+   */
+    width: string,
+};
+
+function HiddenJs(props: JsProps): ?Element<any> {
+  const {
+    children,
+    component,
+    xsUp, // eslint-disable-line no-unused-vars
+    smUp, // eslint-disable-line no-unused-vars
+    mdUp, // eslint-disable-line no-unused-vars
+    lgUp, // eslint-disable-line no-unused-vars
+    xlUp, // eslint-disable-line no-unused-vars
+    xsDown, // eslint-disable-line no-unused-vars
+    smDown, // eslint-disable-line no-unused-vars
+    mdDown, // eslint-disable-line no-unused-vars
+    lgDown, // eslint-disable-line no-unused-vars
+    xlDown, // eslint-disable-line no-unused-vars
+    width,
+    ...other
+  } = props;
+
+  // workaround: see https://github.com/facebook/flow/issues/1660#issuecomment-297775427
+  const ComponentProp = component || defaultProps.component;
+  let visible = true;
+
+  // determine visibility based on the smallest size up
+  for (let i = 0; i < breakpoints.length; i += 1) {
```

Comment:
We could have been using the `.some()` function of an array (returning something like `shouldBeHiddend`).